pkg/apis/pipeline/v1beta1/resolver_types.go

```diff
@@ -33,15 +33,16 @@ type ResolverRef struct {
 	// "repo" or "path" but the set of params ultimately depends on
 	// the chosen resolver.
 	// +optional
+	// +listType=atomic
 	Resource []ResolverParam `json:"resource,omitempty"`
 }
 
 // ResolverParam is a single parameter passed to a resolver.
 type ResolverParam struct {
 	// Name is the name of the parameter that will be passed to the
 	// resolver.
-	Name string
+	Name string `json:"name,omitempty"`
```

Comment:
for fields with "json:omitempty" added, should they have a "+optional" comment as well?

Reply:
Good question - in these cases, they probably shouldn't be `omitempty` in the first place. I'll clean that up.